app/controllers/share/emails_controller.rb

```diff
@@ -0,0 +1,62 @@
+require 'share_progress'
+
+class Share::EmailsController < Share::SharesController
+  before_filter :find_campaign_page
+
+  def index
+    @variations = Share::Email.where(campaign_page_id: @campaign_page.id)
+    render 'share/index'
+  end
+
+  def new
+    @email = Share::Email.new(
+      subject: @campaign_page.title,
+      body: "#{ActionView::Base.full_sanitizer.sanitize(@campaign_page.content).split('.')[0]} {LINK}"
+    )
+
+    render 'share/new'
+  end
+
+  def edit
+    @email = Share::Email.find params[:id]
+    render 'share/edit'
+  end
+
+  def create
+    ShareProgressVariantBuilder.create(permitted_params, {
+      variant_type: :email,
+      campaign_page: @campaign_page,
+      url: campaign_page_url(@campaign_page)
+    })
+
+    respond_to do |format|
+      format.html { redirect_to campaign_page_share_emails_path(@campaign_page)}
+      format.json { render json: {}, status: :ok }
```

Comment:
How does this handle the error case? What happens if someone submits without `{LINK}`, for example?